### PR TITLE
feat: desugar collection literals to vector literals

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/9117

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Desugar.scala [hunk 1]
```diff
@@ -1315,12 +1315,18 @@ object Desugar {
 
   /**
     * Rewrites [[WeededAst.Expr.ListLit]] (`1 :: 2 :: Nil`) expression into `List.Nil` with `List.Cons`.
+    * If there are over 20 literals we translate it to `Vector.toList(Vector#{1, 2, ...})`.
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Desugar.scala:1318]: You are confusing two cases. THIS should ALWAYS be desugared.

The other case I am talking about is called FCons where you need a more tricky check.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Desugar.scala:1318]: This case is `List#{1, 2, 3}` which is always desugared.

File: main/src/ca/uwaterloo/flix/language/phase/Desugar.scala [hunk 2]
```diff
@@ -1295,13 +1295,26 @@ object Desugar {
 
   /**
     * Rewrites [[WeededAst.Expr.FCons]] (`x :: xs`) into a call to `List.Cons(x, xs)`.
+    * If there are over 20 literals we translate it to `Vector.toList(Vector#{1, 2, ...})`.
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Desugar.scala:1298]: I still think we should leave cons untouched - it's also redundant when we add the weeder recursion check

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Desugar.scala:1298]: Its a footgun for our users. What's the downside?

**Comment by magnus-madsen** [comment]
Feel free to copy from https://github.com/flix/flix/pull/9121 and then we can have this in one PR

File: main/src/ca/uwaterloo/flix/language/phase/Desugar.scala [hunk 3]
```diff
@@ -1295,11 +1297,58 @@ object Desugar {
...
+    *
+    * Additionally, if there are over 20 literals and the FCons sequence does not end with the literal `Nil`,
+    * we translate it to `List.append(Vector.toList(Vector#{literals}), nonLiteral)`.
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Desugar.scala:1303]: 👍

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**